src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,144 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader/Loader';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (todoId: number) => Promise<void>,
+  updateTodoStatus: (todo: Todo, status: boolean) => Promise<void>,
+  updateTodoTitle: (todo: Todo, title: string) => Promise<void>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+}) => {
+  const [inputText, setInputText] = useState(todo.title);
+  const [isChangeTitle, setIsChangeTitle] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const titleRef = useRef<HTMLInputElement>(null);
+
+  const toDefaultTitle = () => {
```

Comment:
Let's get used to name the functions correctly according to these rules:
![image](https://user-images.githubusercontent.com/96409563/208611326-a6b0aa16-7331-4cb5-a3bd-3e70dc728cc0.png)